diff --git a/src/vulkan/wsi/wsi_common_drm.c b/src/vulkan/wsi/wsi_common_drm.c
--- a/src/vulkan/wsi/wsi_common_drm.c
+++ b/src/vulkan/wsi/wsi_common_drm.c
@@ -34,4 +34,4 @@
-#define WSI_PRIME_LINEAR_STRIDE_ALIGN 256
+#define WSI_PRIME_LINEAR_STRIDE_ALIGN 32

diff --git a/src/virtio/vulkan/vn_android.c b/src/virtio/vulkan/vn_android.c
--- a/src/virtio/vulkan/vn_android.c
+++ b/src/virtio/vulkan/vn_android.c
@@ -300,6 +300,1 @@ vn_android_gralloc_init(struct vn_device *dev)
-   if (dev->gralloc->type != U_GRALLOC_TYPE_CROS &&
-       dev->gralloc->type != U_GRALLOC_TYPE_IMAPPER) {
-      vn_log(dev->instance, "only CrOS and IMapper v4 grallocs are supported");
-      u_gralloc_destroy(&dev->gralloc);
-      return VK_ERROR_OUT_OF_DATE_KHR;
-   }
+   // Bypassed for NVIDIA compatibility
