diff --git a/src/vulkan/wsi/wsi_common_drm.c b/src/vulkan/wsi/wsi_common_drm.c
--- a/src/vulkan/wsi/wsi_common_drm.c
+++ b/src/vulkan/wsi/wsi_common_drm.c
@@ -37,5 +37,5 @@
 #include <xf86drm.h>
 
-#define WSI_PRIME_LINEAR_STRIDE_ALIGN 256
+#define WSI_PRIME_LINEAR_STRIDE_ALIGN 32
 
 bool

diff --git a/src/virtio/vulkan/vn_android.c b/src/virtio/vulkan/vn_android.c
--- a/src/virtio/vulkan/vn_android.c
+++ b/src/virtio/vulkan/vn_android.c
@@ -170,11 +170,4 @@
    dev->gralloc = gralloc;
 
-   if (dev->gralloc->type != U_GRALLOC_TYPE_CROS &&
-       dev->gralloc->type != U_GRALLOC_TYPE_IMAPPER) {
-      vn_log(dev->instance, "only CrOS and IMapper v4 grallocs are supported");
-      u_gralloc_destroy(&dev->gralloc);
-      return VK_ERROR_OUT_OF_DATE_KHR;
-   }
-
    return VK_SUCCESS;
 }
